### PR TITLE
Make (most) painted parameters scalable.

### PR DESCRIPTION
**Pull request by thorstenhater**
Allow application of `iexpr` scaling onto
- Temperature
- Capacitance
- Resistivity
- Membrane potential
- Ionic parameters: internal/external concentration, diffusivity, and reversal potential.

### CODE REVIEW

File: arbor/include/arbor/cable_cell_param.hpp [hunk 1]
```diff
@@ -96,40 +96,47 @@ struct ARB_SYMBOL_VISIBLE threshold_detector {
 // cell-wide default:
 
 struct ARB_SYMBOL_VISIBLE init_membrane_potential {
-    double value = NAN; // [mV]
+    double value = NAN;         // [mV]
+    std::optional<iexpr> scale; // [()]
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply arbor/include/arbor/cable_cell_param.hpp:100]: Good idea, that's way more convenient.

[the remark this answers is not in the document]
Comment (thorstenhater) [reply arbor/include/arbor/cable_cell_param.hpp:100]: Upon closer inspection this breaks the full architecture of `decor`. I'd rather keep it as is. For now, at least.

File: arbor/fvm_layout.cpp [hunk 2]
```diff
@@ -403,12 +441,28 @@ ARB_ARBOR_API fvm_cv_discretization fvm_cv_discretize(const cable_cell& cell, co
...
-            D.cv_area[i]                 += embedding.integrate_area(c);
-            D.cv_capacitance[i]          += embedding.integrate_area(c.branch, pw_over_cable(capacitance, c, dflt_capacitance));
-            D.init_membrane_potential[i] += embedding.integrate_area(c.branch, pw_over_cable(potential,   c, dflt_potential));
-            D.temperature_K[i]           += embedding.integrate_area(c.branch, pw_over_cable(temperature, c, dflt_temperature));
-            cv_length += embedding.integrate_length(c);
+        for (mcable cable: cv_cables) {
+            auto scale_param = [&](const auto&,
+                                   const auto& par) {
+                if (par.scale) {
+                    auto ie = thingify(*par.scale, provider);
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply arbor/fvm_layout.cpp:448]: We also cannot* use a functor in the outer scope, due to the dependency on `cable` and we cannot* cache the `iexpr_ptr` from `thingify` as we'd need N lambdas.

* without major pains.

File: doc/concepts/decor.rst [hunk 3]
```diff
@@ -194,6 +202,9 @@ Currently, only parameters of :ref:`density mechanisms <cablecell-density-mechs>
     # paint a scaled density mechanism, where 'g' is scaled with the distance from the root.
     decor.paint('"dend"', arbor.scaled_mechanism(arbor.density(m), {'g': '(distance 1.0 (root))'}))
 
+    # initial value for the membrane potential
+    decor.paint('(all)', Vm=(42, '(diameter)'))
```
Comment (AdhocMan) [inline doc/concepts/decor.rst:206]: Is this still correct syntax with the recent changes?

File: doc/python/decor.rst [hunk 4]
```diff
@@ -71,17 +71,19 @@ Cable cell decoration
 
     .. method:: paint(region, Vm=None, cm=None, rL=None, tempK=None)
 
-        Set cable properties on a region.
+        Set cable properties on a region. If a pair of float ``v`` and a string
```
Comment (AdhocMan) [inline doc/python/decor.rst:74]: This should probably be changed to "float or string" as well. Same with the following lines.

**Review by AdhocMan — CHANGES_REQUESTED** [review]
This looks good now. Happy to see you could make the exclusive use of `iexpr` work!
I just got some minor comments on the documentation.

File: arbor/fvm_layout.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdhocMan) [reply arbor/fvm_layout.cpp]: Ok, then this is probably not worth the pain then.

File: doc/concepts/decor.rst (hunk 3 above)
Comment (thorstenhater) [reply doc/concepts/decor.rst:206]: Nope, good catch!

**Review by AdhocMan — APPROVED** [review]

**Pull request merged**